python/ray/autoscaler/_private/event_system.py

```diff
@@ -0,0 +1,25 @@
+from enum import auto, Enum
+
+class CreateClusterEvent(Enum):
+	up_started = auto()
+	ssh_keypair_downloaded = auto()
+	cluster_booting_started = auto()
+	run_initialization_cmd = auto()
+	run_setup_cmd = auto()
+	start_ray_runtime = auto()
+	start_ray_runtime_completed = auto()
+	cluster_booting_completed = auto()
+
+class _EventSystem:
+    def __init__(self):
+        self.callback_map = {}
+
+    def add_callback_handler(self, event, callback):
+        self.callback_map[event] = callback
```

Comment:
Do we want to allow multiple callback handlers? 
i.e. make `self.callback_map: Dict[str, List[callable]]`?

Reply:
+1